app/transformers/PAYETransformer.scala

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package transformers
+
+import models.Amount
+import play.api.Mode.Mode
+import play.api.{Configuration, Play}
+import play.api.libs.json._
+import play.api.libs.json.JsPath
+import uk.gov.hmrc.play.config.ServicesConfig
+
+trait PAYETransformer {
+  def middleTierAmountJson(name: String, value: Double, currency: String = "GBP"): JsObject = Json.obj(
+    name -> Json.obj(
+      "amount"   -> value,
+      "currency" -> currency
+    )
+  )
+
+  def middleTierRateJson(name: String, percent: Double): JsObject = Json.obj(
+    name -> Json.obj(
+      "percent" -> JsString(percent.toString + "%")
+    )
+  )
```

Comment:
This duplicates the functionality in Rate.scala line 40 - further, why are you adding the percentage symbol to the rate? That's entirely a front-end concern and makes this value harder to work with if needed for a calculation elsewhere.